app/javascript/components/ansible-credentials-form/ansible-credential-form.schema.js

```diff
@@ -0,0 +1,37 @@
+import { componentTypes } from '@@ddf';
+
+const createSchema = (fields, promise, edit, loadSchema) => ({
+  fields: [
+    {
+      component: componentTypes.TEXT_FIELD,
+      id: 'name',
+      name: 'name',
+      label: __('Name'),
+      maxLength: 128,
+    },
+    {
+      component: 'select-with-onchange',
+      id: 'type',
+      name: 'type',
+      label: __('Credential type'),
+      placeholder: __('<Choose>'),
+      isDisabled: edit,
+      onChange: value => promise.then(loadSchema(value)),
+      loadOptions: () =>
+        promise.then(
+          ({
+            data: {
+              credential_types: { embedded_ansible_credential_types },
+            },
+          }) =>
+            Object.keys(embedded_ansible_credential_types).map(key => ({
+              value: key,
+              label: __(embedded_ansible_credential_types[key].label),
+            })),
+        ),
+    },
```

Comment:
I guess this field should be also required as we don't want to allow the submit button to be clickable if there's nothing selected in the dropdown.